Firestore/core/src/firebase/firestore/util/firebase_assert.h

```diff
@@ -103,6 +103,11 @@
   FIREBASE_DEV_ASSERT_MESSAGE_WITH_EXPRESSION(expression, expression, \
                                               __VA_ARGS__)
 
+// Indicates an area of the code that cannot be reached (except possibly due to
+// undefined behaviour). The only reasonable thing to do in these cases is to
+// immediately abort.
+#define FIREBASE_UNREACHABLE() abort()
```

Comment:
Alternatively, we could call FailAssert() for a friendlier error message... but since this should be used only for "impossible" conditions, I'm not too concerned about it.